### PR TITLE
Use termporary config in tests

### PR DESCRIPTION
**Pull request by oxitnik**
## Description
Use temporary directory as config directory.

## Checklist

- [x] I've added this contribution to the `changelog.md`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -29,3 +30,8 @@ def executor(connection):
 @pytest.fixture
 def exception_formatter():
     return lambda e: str(e)
+
+
+@pytest.fixture(scope="session", autouse=True)
+def temp_config(tmpdir_factory):
```
Comment (j-bennet) [inline tests/conftest.py]: Where is this fixture being used?

File: tests/features/environment.py [hunk 2]
```diff
@@ -96,6 +98,10 @@ def before_all(context):
 
     context.fixture_data = fixutils.read_fixture_files()
 
+    context.env_config_home = tempfile.mkdtemp(prefix='pgcli_home_')
+
+    os.environ['XDG_CONFIG_HOME'] = context.env_config_home
```
Comment (j-bennet) [inline tests/features/environment.py]: Do you have to restore this variable back to previous value after tests are run?

File: tests/conftest.py (hunk 1 above)
Comment (oxitnik) [reply tests/conftest.py]: It is used automatically on session start

File: tests/features/environment.py (hunk 2 above)
Comment (oxitnik) [reply tests/features/environment.py]: I don't see why. Env vars process-only anyway.

**Comment by j-bennet** [comment]
@oxitnik This looks fine, but it look me some time to understand what's going on. Please add some comments to make it easier for other people working with this code later.

**Comment by oxitnik** [comment]
@j-bennet  Added some comments. Also made XDG_CONFIG_HOME variable to be restored after test.

**Comment by j-bennet** [comment]
Very nice. Thanks!

🍫

**Pull request merged**